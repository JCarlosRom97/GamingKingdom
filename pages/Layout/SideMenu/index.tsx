import React from "react";
import {
  SideBar,
  Title,
  SideBarSearch,
  Button,
  MenuItem,
  SideBarSearchContainer,
  SideBarContent,
  Content,   
  LayoutWrapper,
  LinkMenu
} from "./styles";

import { usePathname } from "next/navigation";

type SideMenuProps = {
  children?: React.ReactElement
}

export default function SideMenu({children}: SideMenuProps) {
  const pathname = usePathname()

  return (
    <LayoutWrapper>
      {/* Side Bar */}
      <SideBar id="sidebar">
        <Title >Gaming Kingdom</Title>
        <SideBarSearchContainer>
          <SideBarSearch
            id="q"
            aria-label="Search contacts"
            placeholder="Search"
            type="search"
            name="q"
          />
          <Button>Go</Button>
        </SideBarSearchContainer>
        <SideBarContent>
          <MenuItem >
            <LinkMenu isactive={pathname === '/' ? 1 : 0} href={'/'}>Home</LinkMenu>
          </MenuItem>
          <MenuItem >
            <LinkMenu isactive={pathname === '/dashboard' ? 1 : 0} href={'/dashboard'}>Dashboard</LinkMenu>
          </MenuItem>
        </SideBarContent>
      </SideBar>
      {/* Content of page */}
      <Content>
        {children}
      </Content>
    </LayoutWrapper>
  );
}
