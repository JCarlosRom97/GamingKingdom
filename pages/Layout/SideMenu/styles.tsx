import styled from "styled-components";
import Link from "next/link";

export const LayoutWrapper = styled.div`
  display: flex;
  flex-direction: row;
  height: 100%;
`;

export const SideBar = styled.div`
  background-color: #9bc0e7;
  padding: 30px;
  width: 20%;
`;

export const Content = styled.div`
  width: 80%;
`;

export const Title = styled.h1`
  padding-bottom: 10px;
  cursor: pointer;
  font-family: "Digital Font";
  color: #606060;
`;

export const SideBarSearchContainer = styled.div`
  display: flex;
  padding-bottom: 10px;
`;

export const SideBarContent = styled.div`
  display: flex;
  flex-direction: column;
`;

export const SideBarSearch = styled.input`
  padding: 3px;
  border-radius: 5px;
  border-width: 1px;
  border-style: solid;
  border-color: #e0e0e0;

  &:focus {
    border-color: #606060;
    outline: none;
    box-shadow: 0px 0px 2px red;
  }
`;

export const Button = styled.button`
  padding: 10px;
  background-color: f6e8d9;
  color: #606060;
  border-radius: 5px;
  border-width: 1px;
  border-style: solid;
  border-color: #e0e0e0;
  cursor: pointer;
`;

export const MenuItem = styled.button`
  background: none;
  color: #606060;
  padding: 10px 0px 10px 0px;
  font-weight: bold;
  text-align: left;
  font-size: 18px;
  border: none;
  cursor: pointer;
  text-decoration: none;
`;

export const LinkMenu = styled(Link)<{props:{isactive:boolean}}>`
  text-decoration: none;
  color: ${props=> props.isactive? 'black': '#606060' } 
`
