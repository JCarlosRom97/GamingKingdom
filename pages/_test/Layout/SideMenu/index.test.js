import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import SideMenu from "../../../Layout/SideMenu";
import { BrowserRouter as Router } from "react-router-dom";
test("renders correctly", () => {
  const { container } = render(
    <Router>
      <SideMenu />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});

test("check links", () => {
  render(
    <Router>
      <SideMenu />
    </Router>,
  );
  const gamingKingdom = screen.getByText("Gaming Kingdom");

  expect(gamingKingdom.textContent).toBe("Gaming Kingdom");

  fireEvent(
    gamingKingdom,
    new MouseEvent("click", {
      bubbles: true,
      cancelable: true,
    }),
  );

  const homeButton = screen.getByText("Home");

  expect(homeButton.textContent).toBe("Home");

  fireEvent(
    homeButton,
    new MouseEvent("click", {
      bubbles: true,
      cancelable: true,
    }),
  );
});
