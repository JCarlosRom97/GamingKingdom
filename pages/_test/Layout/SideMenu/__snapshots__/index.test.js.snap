// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-beyTiQ cKVgyg"
  >
    <div
      class="sc-guDMob evfCrZ"
      id="sidebar"
    >
      <h1
        class="sc-hLQTFJ YRiIl"
      >
        Gaming Kingdom
      </h1>
      <div
        class="sc-eDLKEg bvekgh"
      >
        <input
          aria-label="Search contacts"
          class="sc-gLLuHO jMsPbm"
          id="q"
          name="q"
          placeholder="Search"
          type="search"
        />
        <button
          class="sc-iBdoyZ kDRhHD"
        >
          Go
        </button>
      </div>
      <div
        class="sc-jTQDJr jYBzJp"
      >
        <button
          class="sc-fsYfxw eKksly"
        >
          Home
        </button>
      </div>
    </div>
    <div
      class="sc-dmyDmy lcsCEu"
    />
  </div>
</div>
`;
